Based on patches provided in PR#20725 by Jörg Saßmannshausen 
https://github.com/easybuilders/easybuild-easyconfigs/pull/20725
Update: Cintia Willemyns (Vrije Universiteit Brussel)
--- cellprofiler-4.2.8.orig/setup.py	2024-11-29 16:29:01.573610930 +0100
+++ cellprofiler-4.2.8/setup.py	2024-11-29 16:34:13.898006296 +0100
@@ -66,8 +66,8 @@
         "boto3>=1.12.28",
         "cellprofiler-core==4.2.8",
         "centrosome>=1.2.3,<1.3",
-        "docutils==0.15.2",
-        "h5py~=3.6.0,<4",
+        "docutils>=0.15.2",
+        "h5py>=3.6.0,<4",
         "imageio>=2.5",
         "inflect>=2.1,<7",
         "Jinja2>=2.11.2",
@@ -80,14 +80,14 @@
         "prokaryote==2.4.4",
         "python-bioformats>=4.0.7,<5",
         "python-javabridge>=4.0.3,<5",
-        "pyzmq~=22.3",
-        "sentry-sdk==0.18.0",
+        "pyzmq>=22.3",
+        "sentry-sdk>=0.18.0",
         "requests>=2.22",
-        "scipy==1.9.0",
-        "scikit-image==0.18.3",
+        "scipy>=1.9.0",
+        "scikit-image>=0.18.3",
         "scikit-learn>=0.20,<1",
         "six",
-        "tifffile<2022.4.22",
+        "tifffile==2023.7.18",
         "wxPython>=4.1.0,<5",
     ],
     license="BSD",
